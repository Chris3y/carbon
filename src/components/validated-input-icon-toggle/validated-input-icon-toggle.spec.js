import React from 'react';
import { shallow } from 'enzyme';

import ValidatedInputIconToggle from './validated-input-icon-toggle.component';

const props = {
  iconType: 'foo',
  inputId: '123'
};

describe('ValidatedInputIconToggle', () => {
  let wrapper;

  describe('when initiated without the validationHTML prop', () => {
    beforeEach(() => {
      wrapper = shallow(<ValidatedInputIconToggle { ...props } />);
    });

    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when initiated with the validationHTML prop', () => {
    const mockValidationHTML = <span>mock content</span>;
    const propsWithValidationHTML = { validationHTML: mockValidationHTML, ...props };

    beforeEach(() => {
      wrapper = shallow(<ValidatedInputIconToggle { ...propsWithValidationHTML } />);
    });

    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
