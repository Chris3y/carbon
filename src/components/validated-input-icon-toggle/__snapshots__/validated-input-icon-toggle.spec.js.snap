// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidatedInputIconToggle when initiated with the validationHTML prop renders as expected 1`] = `
<InputIconToggle
  content={
    <span
      className="input-icon-toggle input-icon-toggle--foo"
    >
      <span>
        mock content
      </span>
    </span>
  }
  iconType="foo"
  inputId="123"
/>
`;

exports[`ValidatedInputIconToggle when initiated without the validationHTML prop renders as expected 1`] = `
<InputIconToggle
  content={null}
  iconType="foo"
  inputId="123"
/>
`;
